Modules/Loadable/VolumeRendering/MRML/vtkMRMLShaderPropertyStorageNode.cxx

```diff
@@ -106,7 +106,7 @@ void DecodeMatrix(const Json::Value & json, std::vector<ScalarType> & values)
     for(Json::ArrayIndex columnIndex = 0; columnIndex < nbCols; ++columnIndex)
       {
       ScalarType value;
-      DecodeValue(row[rowIndex], value);
+      DecodeValue(row[columnIndex], value);
```

Comment:
While reviewing the code, I found this issue. it seems the encoding of matrix was incorrect. The `columnIndex` var was not used

Reply:
"add a self-test demonstrating how to perform a shader replacement." 

Can you give an example of an existing test I can use as a reference? Slicer doesn't seem to have the same kind of testing functionality as VTK where test rendering can be compared to an image stored in the repository, am I right?

Reply:
b.t.w. You can check the round-trip test element, it is done.

Reply:
We use VTK's screenshot-based visual check (provided by vtkTesting class), see for example vtkMRMLThreeDReformatDisplayableManagerTest1.

Reply:
Working on a test this morning. I'm trying to mimic what is done in vtkMRMLThreeDReformatDisplayableManagerTest1, but the test image is not generated. It seems like I'm missing something, but not sure what. The output of the vtkMRMLThreeDReformatDisplayableManagerTest1 test has those lines:
```
164: <DartMeasurement name="ImageError" type="numeric/double">0</DartMeasurement><DartMeasurement name="BaselineImage" type="text/string">Standard</DartMeasurement><DartMeasurement name="WallTime" type="numeric/double">0.0364511</DartMeasurement>
```
Which are not present when I run my own test. Any idea what might be missing?

Reply:
Forget it. I found the problem. In CMake, simple_test() doesn't produce an output, you have to use the add_test() macro instead.